neural_lam/utils.py

```diff
@@ -271,3 +275,49 @@ def init_wandb_metrics(wandb_logger):
     experiment.define_metric("val_mean_loss", summary="min")
     for step in constants.VAL_STEP_LOG_ERRORS:
         experiment.define_metric(f"val_loss_unroll{step}", summary="min")
+
+
+@rank_zero_only
+def rank_zero_print(*args, **kwargs):
+    """Print only from rank 0 process"""
+    print(*args, **kwargs)
+
+
+@rank_zero_only
+def init_wandb(args):
+    """Initialize wandb"""
+    if args.resume_run is None:
+        prefix = f"subset-{args.subset_ds}-" if args.subset_ds else ""
+        if args.eval:
+            prefix = prefix + f"eval-{args.eval}-"
+        run_name = (
+            f"{prefix}{args.model}-{args.processor_layers}x{args.hidden_dim}-"
+            f"{time.strftime('%m_%d_%H_%M_%S')}"
```

Comment:
Can we keep an actual random number as part of run name? I have experienced that runs with different configs start at the same second on SLURM, resulting in naming conflicts and checkpoints being over-written. Or is there a better solution to this (having a random number in there is not super nice)?

Reply:
ah yes, I had this issue as well. I think adding a randint is totally appropriate here.